microsoft-edge/extensions-chromium/publish/api/using-addons-publish-api.md

```diff
@@ -0,0 +1,220 @@
+|---|---|
+| _product_ | A single product for which you have an entire entry at Partner Center. |
+| _product ID_ | The product ID of the product whose draft needs to be published.  This is the product ID of your product as shown at Partner Center. |
+| _submission_ | A version of the product.  Every update to a product is a submission, whether draft, in review, or published. |
+| _operation_ | A REST operation such as GET or PUT.  There are several operations, defined at [Add-ons Publish API Reference][AddonsPublishAPIRef]. |
+| _operation ID_ | The ID of a REST operation. |
+
+
+<!-- ====================================================================== -->
+## Before you begin
+
+To use the Microsoft Edge Add-ons Publish API, you need to enable the API for your project in the Microsoft Partner Center.
+
+1. Visit Microsoft Partner Center and login to the account from which you have already published an add-on.
+
+1. Under the **Microsoft Edge** program, a new page for **Publish APIs** appears.
+
+1. Enable the APIs.<!-- what UI action to do that?--> This step may take a few minutes.  After the APIs are enabled, the **Client ID**, **Client Secret**, and **Auth Token URL** are displayed on this page.
+
+1. Note the **ClientID**, **Client Secret** and the **Auth Token URL**.  You'll use them in the next step to get an access token.
+
+
+<!-- ====================================================================== -->
+## Retrieving the access token
+
+After you've acquired the necessary authorization for your application, get access tokens for APIs.  To get a token using the client credentials grant, send a POST request to the Auth Token URL.  The tenant information is available in the URL that you received in the **Before you begin** steps above.
+
+```rest
+Endpoint: https://login.microsoftonline.com/{tenant}/oauth2/v2.0/token
+Type: POST
+Header Parameters: Content-Type: application/x-www-form-urlencoded
+```
+
+### Sample request
+
+```console
+> curl \
+-X POST \
+-H "Content-Type: application/x-www-form-urlencoded" \
+-d "client_id={$Client_ID}" \
+-d "scope=https://addons.edge.microsoft.com/.default" \
+-d "client_secret={$Client_Secret}" \
+-d "grant_type=client_credentials" \
+-v \
+https://login.microsoftonline.com/{tenant}/oauth2/v2.0/token
+```
+
+### Sample response
+
+```json
+{
+  "token_type": "Bearer",
+  "expires_in": 3599,
+  "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Ik1uQ19WWmNBVGZNNXBP..."
+}
+```
+
+For more information, navigate to [OAuth 2.0 client credentials flow on the Microsoft identity platform][AzureOAuthGetToken].
+
+
+<!-- ====================================================================== -->
+## Using the Publish API
+
+Once you have an access token, your app can use the Microsoft Edge Add-ons Publish API.  There are endpoints for getting a list of products, updating a product, and publishing a product.<!-- correct list of actions? there wasn't a get-list section below, only in API Ref. -->
```

Comment:
Suggested wording:

Once you have an access token, you can use the Microsoft Edge Add-ons Publish API. The API exposes endpoints for getting a list of products, updating, and publishing them.